From 9dbaffbd83e8937ce55537caec339952d4804491 Mon Sep 17 00:00:00 2001
From: Arunachalam Ganapathy <arunachalam.ganapathy@arm.com>
Date: Fri, 5 Apr 2024 15:29:07 +0100
Subject: [PATCH] fix: unused function md_type_to_string

This fixes unused function md_type_to_string when compiled with LLVM

Signed-off-by: Arunachalam Ganapathy <arunachalam.ganapathy@arm.com>
---
 library/x509.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/library/x509.c b/library/x509.c
index fc13b9213..bf01c4c5d 100644
--- a/library/x509.c
+++ b/library/x509.c
@@ -132,6 +132,7 @@ int mbedtls_x509_get_alg(unsigned char **p, const unsigned char *end,
 /*
  * Convert md type to string
  */
+#if defined(MBEDTLS_X509_RSASSA_PSS_SUPPORT)
 static inline const char *md_type_to_string(mbedtls_md_type_t md_alg)
 {
     switch (md_alg) {
@@ -169,6 +170,7 @@ static inline const char *md_type_to_string(mbedtls_md_type_t md_alg)
             return NULL;
     }
 }
+#endif /* MBEDTLS_X509_RSASSA_PSS_SUPPORT */
 
 #if defined(MBEDTLS_X509_RSASSA_PSS_SUPPORT)
 /*
-- 
2.39.2

